src/core/lib/surface/call.cc

```diff
@@ -565,6 +571,8 @@ void grpc_call_unref(grpc_call* c) {
   if (GPR_LIKELY(!c->ext_ref.Unref())) return;
 
   GPR_TIMER_SCOPE("grpc_call_unref", 0);
+  intptr_t unrefs_needed =
+      1 + c->batches_completed.Load(grpc_core::MemoryOrder::RELAXED);
```

Comment:
I'm not sure this is correct.  I think there might be cases (especially when cancellation occurs) where the application may call `grpc_call_unref()` while we still have batches outstanding.  If that happens, then we'd wind up with a memory leak, because nothing will ever release the refs for batches that finish after `grpc_call_unref()` is called.

Reply:
Yeah, I think for now it's back to the drawing board. I'll leave this PR open for now but am not sure where else we can find a sync'd point for free like here.